lib/features/allow_ptrace.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * This feature adjusts the AppArmor profile to allow ptrace within a container.
+ * All of the interesting action takes place in `lib/task.js`.
+ */
+
+// Prefix used in scope matching for docker-worker features
+const FEATURE_SCOPE_PREFIX = 'docker-worker:feature:';
+
+export default class AllowPtrace {
+  constructor() {
+    this.featureName = 'allowPtrace';
+  }
+
+  async link(task) {
+    let docker = task.runtime.docker;
```

Comment:
I'm working on writing and running some tests right now, so I'll catch that
kind of thing.

On Fri, Dec 11, 2015 at 9:39 AM, Greg Arndt <notifications@github.com>
wrote:

> In lib/features/allow_ptrace.js
> <https://github.com/taskcluster/docker-worker/pull/192#discussion_r47360795>
> :
>
> > @@ -0,0 +1,24 @@
> > +/**
> > + * This feature adjusts the AppArmor profile to allow ptrace within a container.
> > + * All of the interesting action takes place in `lib/task.js`.
> > + */
> > +
> > +// Prefix used in scope matching for docker-worker features
> > +const FEATURE_SCOPE_PREFIX = 'docker-worker:feature:';
> > +
> > +export default class AllowPtrace {
> > +  constructor() {
> > +    this.featureName = 'allowPtrace';
> > +  }
> > +
> > +  async link(task) {
> > +    let docker = task.runtime.docker;
>
> not using docker here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskcluster/docker-worker/pull/192/files#r47360795>.
>